src/utils/moderation.rs

```diff
@@ -93,18 +93,21 @@ pub fn queue_unmute_member(
             .await?
             .ok_or("User was not muted.")?;
 
-        let taken_roles = db_result
-            .taken_roles
-            .unwrap()
```

Comment:
So basically this was the culprit. What causes it to be `None`?

Reply:
> So basically this was the culprit. What causes it to be `None`?

https://github.com/revanced/revanced-discord-bot/blob/6b08dfd68cffd9245b11e09e1f11ff52719240d2/src/commands/moderation.rs#L279-L288

`taken_roles` is only ever initially set if the target is a member.